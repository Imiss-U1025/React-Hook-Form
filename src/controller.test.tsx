import * as React from 'react';
import { render } from '@testing-library/react';
import { Controller } from './index';

describe('React Hook Form Input', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Controller
        name="test"
        as="input"
        control={{
          defaultValues: {},
          setValue: jest.fn(),
          register: jest.fn(),
          unregister: jest.fn(),
          errors: {},
          mode: { isOnSubmit: false, isOnBlur: false },
          reValidateMode: {
            isReValidateOnBlur: false,
            isReValidateOnSubmit: false,
          },
          formState: { isSubmitted: false },
        }}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
