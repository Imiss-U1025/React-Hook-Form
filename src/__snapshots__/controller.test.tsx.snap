// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller should support custom value name 1`] = `
<DocumentFragment>
  <input
    value=""
  />
</DocumentFragment>
`;

exports[`Controller should support default value from hook form 1`] = `
<DocumentFragment>
  <input
    value="data"
  />
</DocumentFragment>
`;
